src/utils/use-do-once-when.ts

```diff
@@ -1,19 +1,27 @@
 import {useEffect, useRef} from 'react';
 
-export function useDoOnceWhen(fn: () => void, condition: boolean) {
-  const firstTimeRef = useRef(true);
+export function useDoOnceWhen(
+  fn: () => void,
+  condition: boolean,
+  onlyOnce?: boolean,
+) {
+  const isFirstTimeRef = useRef(true);
   const fnRef = useRef(fn);
+
   useEffect(() => {
     fnRef.current = fn;
   }, [fn]);
 
   useEffect(() => {
-    if (firstTimeRef.current && condition) {
-      firstTimeRef.current = false;
+    if (isFirstTimeRef.current && condition) {
       fnRef.current();
+      isFirstTimeRef.current = false;
     }
+
     return () => {
-      firstTimeRef.current = true;
+      if (!onlyOnce) {
+        isFirstTimeRef.current = true;
+      }
     };
-  }, [condition]);
+  }, [condition, onlyOnce]);
```

Comment:
Also add some tests on what happens when `onlyOnce` changes, since it is used in the dependency array for the main `useEffect` in the util.